### PR TITLE
Apply macro expansion based on MutateInOpts

### PR DESCRIPTION
**Pull request by adamcfraser**
Uptake change in sg-bucket to provide the required macro expansions in a MutateInOpts parameter for the relevant methods, instead of hardcoding them into the implementations.

Currently the sg-bucket API only supports a single xattr per operation - will be expanded to multiple xattrs in a subsequent enhancement.

### CODE REVIEW

File: collection+subdoc.go
Comment (torcolvin) [inline collection+subdoc.go:153]: readability nit:

I'd name `numLeaves := len(path) -1` because we use it again.

File: collection+subdoc.go
Comment (torcolvin) [inline collection+subdoc.go:158]: Can we map this a compile time error by requiring map[string]any in compiler string?

File: collection+xattrs.go
Comment (torcolvin) [inline collection+xattrs.go:584]: nit: should this use `fmt.Stringer` ?

File: collection+xattrs.go
Comment (torcolvin) [inline collection+xattrs.go:601]: can we make this a compile time error? and check for nil here?

File: collection+xattrs.go
Comment (torcolvin) [inline collection+xattrs.go:614]: Should we list path[0] in this error, and maybe the xattr we expect?

File: collection+xattrs.go
Comment (torcolvin) [inline collection+xattrs.go:623]: should this be path[1:] that is logged here?

Can you wrap the error here?

**Review by torcolvin — COMMENTED** [review]
These are largely nitpick comments for test code, but might help us debug things in the future. Feel free to disregard.

File: collection+subdoc.go
Comment (adamcfraser) [reply collection+subdoc.go:153]: I evaluated this, but prefer leaving as len(path)-1 as I feel `lastPath := path[len(path)-1]` is more readable than having to go back to see what numLeaves is.

File: collection+subdoc.go
Comment (adamcfraser) [reply collection+subdoc.go:158]: This is required to support non-map JSON.

File: collection+xattrs.go
Comment (adamcfraser) [reply collection+xattrs.go:584]: Intentionally avoided this as I think we'd prefer to report CAS normally when logging, and only use this format for compatibility with CBS.

File: collection+xattrs.go
Comment (adamcfraser) [reply collection+xattrs.go:601]: As above, required for generic JSON support.

File: collection+xattrs.go
Comment (adamcfraser) [reply collection+xattrs.go:614]: Agree - fixed.

File: collection+xattrs.go
Comment (adamcfraser) [reply collection+xattrs.go:623]: I think full path is preferable for context, but agree on wrapping the error.

**Review by torcolvin — APPROVED** [review]
LGTM, will need to bump sg-bucket once the merge happens.

**Review by torcolvin — APPROVED** [review]

**Pull request merged**
